api/features/serializers.py

```diff
@@ -270,11 +271,32 @@ def get_last_modified_in_current_environment(
         return getattr(instance, "last_modified_in_current_environment", None)
 
 
-class UpdateFeatureSerializer(ListCreateFeatureSerializer):
-    """prevent users from changing certain values after creation"""
+class FeatureSerializerWithMetadata(
+    SerializerWithMetadata, ListCreateFeatureSerializer
+):
+    metadata = MetadataSerializer(required=False, many=True)
 
     class Meta(ListCreateFeatureSerializer.Meta):
-        read_only_fields = ListCreateFeatureSerializer.Meta.read_only_fields + (
+        fields = ListCreateFeatureSerializer.Meta.fields + ("metadata",)
+
+    def get_project(self, validated_data: dict = None) -> Project:
+        try:
+            project = self.context.get("project")
+            if project:
+                return project
+            else:
+                raise serializers.ValidationError("Project not found.")
+        except Exception as exc:
```

Comment:
Don't think we need this try/except any more?

Reply:
Removed.